tests/NuGet.Jobs.Catalog2Registration.Tests/Hives/HiveMergerFacts.Support.cs

```diff
@@ -0,0 +1,183 @@
+
+                Config = new Catalog2RegistrationConfiguration
+                {
+                    MaxLeavesPerPage = 3,
+                };
+                Options.Setup(x => x.Value).Returns(() => Config);
+
+                Target = new HiveMerger(
+                    Options.Object,
+                    Logger);
+            }
+
+            public Mock<IOptionsSnapshot<Catalog2RegistrationConfiguration>> Options { get; }
+            public RecordingLogger<HiveMerger> Logger { get; }
+            public Catalog2RegistrationConfiguration Config { get; }
+            public HiveMerger Target { get; }
+        }
+
+        private class VersionAction
+        {
+            public VersionAction(NuGetVersion version, bool isDelete)
+            {
+                Version = version;
+                IsDelete = isDelete;
+            }
+
+            public NuGetVersion Version { get; }
+            public bool IsDelete { get; }
+        }
+
+        private class TestCase
+        {
+            public TestCase(List<NuGetVersion> existing, List<VersionAction> updated)
+            {
+                Existing = existing;
+                Updated = updated;
+            }
+
+            public List<NuGetVersion> Existing { get; }
+            public List<VersionAction> Updated { get; }
+        }
+
+        private static VersionAction Details(string version)
+        {
+            return new VersionAction(NuGetVersion.Parse(version), isDelete: false);
+        }
+
+        private static VersionAction Delete(string version)
+        {
+            return new VersionAction(NuGetVersion.Parse(version), isDelete: true);
+        }
+
+        private static string[] GetVersionArray(HashSet<LeafInfo> leafInfos)
+        {
+            return leafInfos
+                .OrderBy(x => x.Version)
+                .Select(x => x.LeafItem.CatalogEntry.Version)
+                .ToArray();
+        }
+
+        private static async Task<string[]> GetVersionArrayAsync(IndexInfo indexInfo)
+        {
+            var versions = await GetVersionsAsync(indexInfo);
+            return versions.Select(x => x.ToNormalizedString()).ToArray();
+        }
+
+        private static async Task<List<NuGetVersion>> GetVersionsAsync(IndexInfo indexInfo)
+        {
+            var versions = new List<NuGetVersion>();
+            foreach (var pageInfo in indexInfo.Items)
+            {
+                var leafInfos = await pageInfo.GetLeafInfosAsync();
+                foreach (var leafInfo in leafInfos)
+                {
+                    versions.Add(leafInfo.Version);
+                }
+            }
+
+            return versions;
+        }
+
+        private static List<CatalogCommitItem> MakeSortedCatalog(params VersionAction[] versions)
+        {
+            return MakeSortedCatalog(versions.ToList());
+        }
+
+        private static List<CatalogCommitItem> MakeSortedCatalog(List<VersionAction> sortedVersionActions)
```

Comment:
`ICollection<VersionAction>`? Will save the `ToList()` call couple of lines above.

Reply:
Done.